### ISSUE TITLE
TotalFreedomMod API

### ISSUE THREAD

**Issue by ghost** [TotalFreedom/TotalFreedomMod#1990]
Well, you look in Essentials, Worldedit, really, any popular plugin has an API, or most commonly atleast some events. I am currently investigating and adding an event API into the TFM, it is more simple then you think. I will keep this thread updated.
**UPDATE: onAdminAdded and onAdminRemoved events work.** 
Please comment any events that would be useful to add.

**Comment by Commodore64x** [comment]
UPDATE: onAdminAdded and onAdminRemoved events work.

**Comment by Wild1145** [comment]
Sounds good, if people can create a more through idea of what the scope of this task will be we can accept it.

**Comment by ghost** [comment]
@Wild1145 a more through idea? Do you mean thorough? Anyway, the basic scope is this: even thought the TFM is open source, it helps, especially with edited versions of various plugins floating around that some functions in the TFM can be accessed through events such as onAdminAdded.

**Comment by Wild1145** [comment]
@Commodore64x Yes, thats a mix of auto correct and Dyslexia there... What I mean is a list of the events for example you want to be available, how you're going to restrict API access to only authorised applications. These types of things is what we need to know before we can really go "Yeah, thats a great idea"

**Comment by ghost** [comment]
@Wild1145 I was thinking along these lines:
onAdminAdded
onAdminRemoved
onAdminRankChanged
onPlayerPunished
And a few more.
As for restricting it, whatever for? It isn't like a HTTPD module that could contain sensitive information, and I don't know of a single plugin who has an event API and 'restricts' it to a certain plugin...

**Comment by Wild1145** [comment]
@Commodore64x I dont think we're thinking the same thing. The idea for me of an API is that external plugins could interact with them if they wanted. Currently I dont really see the value to just chucking in a load of events, they would need to have some sort of a system whereby other plugins could for example add a warning to the amount of warnings, or ban a player through TFM etc etc.

**Comment by ghost** [comment]
@Wild1145 This is only 1 out of many APIs that should (and could) be added. Server info in JSON format, what you mentioned, and events. This is not the be-all and end-all but it is a good start.

**Comment by Wild1145** [comment]
This issue has been superseded by #2038
